samvera-labs/bulkrax
# 591 export from one tenant into another

ref: #591 

# expected behavior
- we can export "all" from tenant A, and then import that zip file into tenant B and it be like a duplication of tenant A (maintaining relationships, record ids, etc.)

# demo
- I tested by exporting all from the "dev" tenant as a csv([export_2_2_1.zip](https://github.com/samvera-labs/bulkrax/files/9264070/export_2_2_1.zip)) and bags([export_1_1_1.zip](https://github.com/samvera-labs/bulkrax/files/9264069/export_1_1_1.zip)), then importing the csv file in a new "dev2" tenant and the bagit file into a new "dev3" tenant

| | original tenant | new tenant |
| --- | --- | --- |
| csv | ![image](https://user-images.githubusercontent.com/29032869/182969848-b04f30c1-009c-4314-bc78-e64a33b16dc2.png) | ![image](https://user-images.githubusercontent.com/29032869/182969917-f125adba-bcd2-4ab2-98b5-e4e42fb4f15c.png) |
| | ![Screen Shot 2022-08-04 at 6 07 36 PM](https://user-images.githubusercontent.com/29032869/182970086-b467ef43-4083-41e0-a26e-1629f004914b.png) | ![Screen Shot 2022-08-04 at 6 09 17 PM](https://user-images.githubusercontent.com/29032869/182970271-0e73ce64-e33e-44f9-80fc-b52bb6b9c6f1.png) |
| bag | ![image](https://user-images.githubusercontent.com/29032869/182969730-03b31757-05db-4be2-901a-d9e6e5121df6.png) | ![image](https://user-images.githubusercontent.com/29032869/182969800-de0c6494-2360-40b4-bac6-689b2b7ee98d.png)|
|  | ![Screen Shot 2022-08-04 at 5 53 52 PM](https://user-images.githubusercontent.com/29032869/182969066-3f72363b-10f9-425b-b981-18ad47f65862.png) | ![Screen Shot 2022-08-04 at 5 58 13 PM](https://user-images.githubusercontent.com/29032869/182969307-f96c17ea-1abb-4efd-a2f1-82d0fe9a94ae.png) |

# testing
- have some works, file sets and collections in a tenant
- create a new exporter and export "all" with metadata and files using the csv or bagit parsers
- create a new tenant
- create a new importer using the csv or bagit parsers and upload the corresponding exported zip
- check the two tenants against each other

_caveat: work records are the main focus for bagit exporting. if a collection or file set exists in the first tenant and it doesn't have a relationship with a work, it will NOT be exported using the bagit exporter_
